web-app/src/app/stitch-pattern-card.html

```diff
@@ -87,23 +86,18 @@
         this.set('route.path', '/marker');
       }
 
-      patternInfoChanged(patternInfo) {
-        fetch(SM.apiUrl + patternInfo.links.find(link => link.rel === 'thumbnail').href,
-          {
-            headers: window.getAuthHeaders()
-          }).then((response) => {
-            response.arrayBuffer().then((buffer) => {
-              this.imageUrl = 'data:image/png;base64,' + btoa(String.fromCharCode(...new Uint8Array(buffer)));
-            });
-          });
+      async patternInfoChanged(patternInfo) {
+        const response = await window.apiGetThumbnail(patternInfo);
```

Comment:
`apiGetThumbnail` should not be implemented in global context. This function belong to pattern card component only. Window should contain generic `get`, `post`, `delete` function following the analogy with `fetch`.